wrapanapi/systems/ec2.py

```diff
@@ -1451,3 +1452,22 @@ def create_stack(self, name, template_url=None, template_body=None, parameters=N
         stack_id = response.get('StackId')
         return CloudFormationStack(system=self, uuid=stack_id,
                                    raw=self.cloudformation_resource.Stack(stack_id))
+
+    def set_sns_topic_target_for_all_cw_rules(self, topic_arn):
+        # After recreating sns topic cloudwatch rule targets are not set so we need to set them back
+        try:
+            # Get all enabled rules
+            rules = self.cw_events_connection.list_rules().get('Rules')
+            enabled_rules = []
+            for rule in rules:
+                if rule.get('State') == "ENABLED":
+                    enabled_rules.append(rule.get('Name'))
+            # Set targets to rules again
+            for enabled_rule in enabled_rules:
+                target = self.cw_events_connection.list_targets_by_rule(
+                    Rule=enabled_rule).get('Targets')[0]
+                target['Arn'] = topic_arn
+                self.cw_events_connection.put_targets(Rule=enabled_rule, Targets=[target])
+                return True
```

Comment:
This doesn't look right - why return within the for loop? Do you only want to enable the first rule in the list?